web/src/pages/api/public/ingestion.ts

```diff
@@ -297,35 +326,28 @@ export default async function handler(
         const queuePayload: LegacyIngestionEventType =
           env.LANGFUSE_S3_EVENT_UPLOAD_ENABLED === "true" && !s3UploadErrored
             ? {
-                data: sortedBatch.map((event) => ({
-                  type: event.type,
-                  eventBodyId: event.body.id ?? "",
-                  eventId: event.id,
-                })),
+                data: Object.keys(sortedBatchByEventBodyId).map(
+                  (eventBodyId) => {
+                    const { key, type } = sortedBatchByEventBodyId[eventBodyId];
+                    return {
+                      type,
+                      eventBodyId,
+                      eventId: key,
+                    };
+                  },
+                ),
                 authCheck,
                 useS3EventStore: true,
               }
             : { data: sortedBatch, authCheck, useS3EventStore: false };
 
         try {
-          await queue.add(
-            QueueJobs.LegacyIngestionJob,
-            {
-              payload: queuePayload,
-              id: randomUUID(),
-              timestamp: new Date(),
-              name: QueueJobs.LegacyIngestionJob as const,
-            },
-            {
-              removeOnFail: 1_000_000,
-              removeOnComplete: true,
-              attempts: 5,
-              backoff: {
-                type: "exponential",
-                delay: 1000,
-              },
-            },
```

Comment:
This is actually covered by the default settings on the queue. No need to repeat that here for each event.